Batteries/Data/List/Count.lean

```diff
@@ -25,8 +26,44 @@ namespace List
 
 section count
 
-variable [DecidableEq α]
+theorem count_singleton' [DecidableEq α] (a b : α) : count a [b] = if b = a then 1 else 0 :=
+  count_singleton.trans (by grind)
 
-theorem count_singleton' (a b : α) : count a [b] = if b = a then 1 else 0 := by simp [count_cons]
+variable [BEq α]
```

Comment:
If `variable` is really needed then it should be confined to a section, preferably named or otherwise documented.

Reply:
Why isn't it?